pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,569 @@
+package reporting
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/operator-framework/operator-metering/pkg/operator/prestostore"
+
+	"github.com/operator-framework/operator-metering/pkg/apis/metering/v1alpha1"
+	metering "github.com/operator-framework/operator-metering/pkg/apis/metering/v1alpha1"
+	"github.com/operator-framework/operator-metering/pkg/presto"
+	"github.com/operator-framework/operator-metering/test/testhelpers"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestRenderQuery(t *testing.T) {
+	const (
+		testNamespace            = "default"
+		testReportName           = "unready-deployment-replicas"
+		testReportQuery          = "unready-deployment-replicas"
+		testInvalidQueryName     = "invalid-query"
+		testNonExistentQueryName = "does-not-exist"
+		testCatalogName          = "hive"
+		testSchemaName           = "default"
+		testTimeString           = "2019-05-01T15:00:05Z"
+	)
+
+	reportStart := &time.Time{}
+	reportEndTmp := reportStart.AddDate(0, 1, 0)
+	reportEnd := &reportEndTmp
+
+	ds1 := testhelpers.NewReportDataSource("datasource1", testNamespace)
+	ds1.Status.TableRef = v1.LocalObjectReference{Name: "initialized_datasource"}
+	ds2 := testhelpers.NewReportDataSource("datasource2", testNamespace)
+
+	testValidQuery := &metering.ReportQuery{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testReportQuery,
+			Namespace: testNamespace,
+		},
+		Spec: metering.ReportQuerySpec{
+			Query: "SELECT * FROM test_table",
+		},
+	}
+
+	testTable := []struct {
+		name            string
+		reportTemplate  *ReportQueryTemplateContext
+		templateContext TemplateContext
+		expectErr       bool
+		expectErrMsg    string
+		expectOutput    string
+	}{
+		{
+			name: "valid report query with no templating returns nil and the expected query output",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery: testValidQuery,
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{},
+			},
+			expectOutput: "SELECT * FROM test_table",
+		},
+		{
+			name: "valid report query with valid templating returns nil and the expected query output",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
```

Comment:
It looks like most of these are using `testReportName` but should be using `testReportQuery` for the first arg